lib/manifoldco_signature/signature.rb

```diff
@@ -0,0 +1,56 @@
+require 'base64url'
+require 'rbnacl/libsodium'
+
+module ManifoldcoSignature
+
+  class Signature
+
+    def initialize(req)
+      sig = req.env['HTTP_X_SIGNATURE'].split
```

Comment:
It's all pretty dumb right now. well, it's correct, but not informative.

if anything is wrong with the b64 of the sigs/keys, or headers are missing, or what have you, the code will throw, and be caught above and treated as a failed verification.

Given that from a provider's POV all they'll be able to do is misconfigure the key, and we will throw if that happens and its not a valid key, that's really enough.